src/transformers/models/umt5/modeling_umt5.py

```diff
@@ -223,23 +199,61 @@ def _relative_position_bucket(self, relative_position):
         log_ratio = log_ratio * (num_buckets - max_exact)
         relative_position_if_large = max_exact + log_ratio.to(torch.long)
         relative_position_if_large = torch.min(
-            relative_position_if_large, torch.full_like(relative_position_if_large, num_buckets - 1)
+            relative_position_if_large, torch.full_like(relative_position_if_large, num_buckets - 1, device=self.weight.device)
         )
 
         relative_buckets += torch.where(is_small, relative_position, relative_position_if_large)
         return relative_buckets
 
-    def compute_bias(self, query_length, key_length, device=None):
+    def _set_cached_bias(self, query_length, key_length):
         """Compute binned relative position bias"""
-        if device is None:
-            device = self.relative_attention_bias.weight.device
-        context_position = torch.arange(query_length, dtype=torch.long, device=device)[:, None]
-        memory_position = torch.arange(key_length, dtype=torch.long, device=device)[None, :]
+        context_position = torch.arange(query_length, dtype=torch.long, device = self.weight.device)[:, None]
+        memory_position = torch.arange(key_length, dtype=torch.long, device = self.weight.device)[None, :]
         relative_position = memory_position - context_position  # shape (query_length, key_length)
         relative_position_bucket = self._relative_position_bucket(relative_position)
-        values = self.relative_attention_bias(relative_position_bucket)  # shape (query_length, key_length, num_heads)
-        values = values.permute([2, 0, 1]).unsqueeze(0)  # shape (1, num_heads, query_length, key_length)
-        return values
+        values = super().forward(relative_position_bucket).to(self.weight.device) # shape (query_length, key_length, num_heads)
+        # shape (1, num_heads, query_length, key_length)
+        cached_bias = values.permute([2, 0, 1]).unsqueeze(0)
+        self.register_buffer("cached_bias", cached_bias, persistent=False)
```

Comment:
When we use `device_map`, the non persistent buffer is on the meta device (comment above). Moreover, as the value is not present in the `state dict`,  `cached_bias` will stay on the meta device when we are loading the weights . Then, when we try to dispatch the model to cuda, the following error will appear : `Cannot copy out of meta tensor; no data!`